hs_geo_raster_resource/migrations/custom_migration_for_raster_meta_update_20160512.py

```diff
@@ -0,0 +1,147 @@
+import subprocess
+import xml.etree.ElementTree as ET
+
+from django.db import migrations
+from django.core.files.uploadedfile import UploadedFile
+
+from django_irods.storage import IrodsStorage
+from hs_core import hydroshare
+from hs_core.hydroshare.utils import resource_modified
+from hs_geo_raster_resource.models import RasterResource
+from hs_geo_raster_resource import raster_meta_extract
+
+
+def migrate_tif_file(apps, schema_editor):
+    log = logging.getLogger()
+    istorage = IrodsStorage()
+    success = 0
+    fail = []
+    not_modify = []
+
+    for res in RasterResource.objects.all():
+        if res.files.all():
+            # update metadata information
+            try:
+                # copy all the files of a resource to temp dir
+                temp_dir = tempfile.mkdtemp()
+                for res_file in res.files.all():
+                    shutil.copy(res_file.resource_file.file.name,
+                                os.path.join(temp_dir, os.path.basename(res_file.resource_file.name)))
+
+                # update vrt file for single tif file
+                if len(os.listdir(temp_dir)) == 2:
+                    # create new vrt file
+                    tif_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.tif' == f[-4:]].pop()
+                    vrt_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.vrt' == f[-4:]].pop()
+
+                    try:
+                        with open(os.devnull, 'w') as fp:
+                            subprocess.Popen(['gdal_translate', '-of', 'VRT', tif_file_path, vrt_file_path], stdout=fp, stderr=fp).wait()   # remember to add .wait()
+
+                        # modify the vrt file contents
+                        tree = ET.parse(vrt_file_path)
+                        root = tree.getroot()
+                        for element in root.iter('SourceFilename'):
+                            element.attrib['relativeToVRT'] = '1'
+                        tree.write(vrt_file_path)
+                        
+                        # delete vrt res file
+                        for f in res.files.all():
+                            if 'vrt' == f.resource_file.name[-3:]:
+                                f.resource_file.delete()
+                                f.delete()
+
+                        # add new vrt file to resource
+                        new_file = UploadedFile(file=open(vrt_file_path, 'r'), name=os.path.basename(vrt_file_path))
+                        hydroshare.add_resource_files(res.short_id, new_file)
+
+                        # update the bag
+                        bag_name = 'bags/{res_id}.zip'.format(res_id=res.short_id)
+                        if istorage.exists(bag_name):
+                            # delete the resource bag as the old bag is not valid
+                            istorage.delete(bag_name)
+                        resource_modified(res, res.creator)
+
+                    except:
```

Comment:
@jbradberry Good catch and good point. @gantian127 I missed this during code review. When looking at the code again, I don't see an need for this inner try...except block included in an outer try..except block. You can print out exception message along with stack trace so that you know which line failed during migration without using an inner try..except block for vrt file update just to print out a failure message for vrt update. I think the inner try...except can be removed while adding printing full stack trace in the outer try...except block.